commit 27eaac2726508999231ac8591c1833d9356f02c9
Author: Roopa Prabhu <roopa@cumulusnetworks.com>
Date:   Wed Oct 11 14:21:24 2017 -0700

    route: neigh: print family in neigh dumps

    Since the neigh cache can contain msgs from different families,
    print family id to distinguish neigh cache entries in the
    dump output

    Signed-off-by: Roopa Prabhu <roopa@cumulusnetworks.com>
    Signed-off-by: Thomas Haller <thaller@redhat.com>

commit 75ea0bee778d010f20e685bc4780b84d4139ab9e
Author: Roopa Prabhu <roopa@cumulusnetworks.com>
Date:   Wed Oct 11 14:21:23 2017 -0700

    route: link: add family to dump messages

    Since the link cache can contain msgs from different families,
    print family id to distinguish link cache entries in the
    dump output

    Signed-off-by: Roopa Prabhu <roopa@cumulusnetworks.com>
    Signed-off-by: Thomas Haller <thaller@redhat.com>

--- a/lib/route/link.c
+++ b/lib/route/link.c
@@ -831,6 +831,9 @@ static void link_dump_line(struct nl_obj
 		fetched_cache = 1;
 	}
 
+	if (link->l_family != AF_UNSPEC)
+		nl_dump_line(p, "%s ", nl_af2str(link->l_family, buf, sizeof(buf)));
+
 	nl_dump_line(p, "%s %s ", link->l_name,
 		     nl_llproto2str(link->l_arptype, buf, sizeof(buf)));
 
--- a/lib/route/neigh.c
+++ b/lib/route/neigh.c
@@ -462,10 +462,14 @@ static void neigh_dump_line(struct nl_ob
 	struct rtnl_neigh *n = (struct rtnl_neigh *) a;
 	struct nl_cache *link_cache;
 	char state[128], flags[64];
+	char buf[128];
 
 	link_cache = nl_cache_mngt_require_safe("route/link");
 
-	if (n->n_family != AF_BRIDGE)
+	if (n->n_family != AF_UNSPEC)
+		nl_dump_line(p, "%s ", nl_af2str(n->n_family, buf, sizeof(buf)));
+
+	if (n->ce_mask & NEIGH_ATTR_DST)
 		nl_dump_line(p, "%s ", nl_addr2str(n->n_dst, dst, sizeof(dst)));
 
 	if (link_cache)
